uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -590,158 +739,268 @@ protected AliasFeatureDisabled() {
+                    final IdentityProvider<?> originalIdp = executeWithTemporarilyEnabledAliasFeature(
+                            aliasFeatureEnabled,
+                            () -> createIdpWithAlias(zone1, zone2)
+                    );
+
+                    // change non-alias property without setting alias properties to null
+                    originalIdp.setName("some-new-name");
+                    shouldRejectUpdate(zone1, originalIdp, HttpStatus.UNPROCESSABLE_ENTITY);
+                }
 
-            @Test
-            void shouldAccept_SetOnlyAliasPropertiesToNull_CustomToUaaZone() throws Throwable {
-                shouldAccept_SetOnlyAliasPropertiesToNull(customZone, IdentityZone.getUaa());
-            }
+                @Test
+                void shouldAccept_SetOnlyAliasPropertiesToNull_UaaToCustomZone() throws Throwable {
+                    shouldAccept_SetOnlyAliasPropertiesToNull(IdentityZone.getUaa(), customZone);
+                }
 
-            private void shouldAccept_SetOnlyAliasPropertiesToNull(
-                    final IdentityZone zone1,
-                    final IdentityZone zone2
-            ) throws Throwable {
-                final IdentityProvider<?> originalIdp = executeWithTemporarilyEnabledAliasFeature(
-                        aliasFeatureEnabled,
-                        () -> createIdpWithAlias(zone1, zone2)
-                );
+                @Test
+                void shouldAccept_SetOnlyAliasPropertiesToNull_CustomToUaaZone() throws Throwable {
+                    shouldAccept_SetOnlyAliasPropertiesToNull(customZone, IdentityZone.getUaa());
+                }
+
+                private void shouldAccept_SetOnlyAliasPropertiesToNull(
+                        final IdentityZone zone1,
+                        final IdentityZone zone2
+                ) throws Throwable {
+                    final IdentityProvider<?> originalIdp = executeWithTemporarilyEnabledAliasFeature(
+                            aliasFeatureEnabled,
+                            () -> createIdpWithAlias(zone1, zone2)
+                    );
+
+                    final String initialAliasId = originalIdp.getAliasId();
+                    assertThat(initialAliasId).isNotBlank();
+                    final String initialAliasZid = originalIdp.getAliasZid();
+                    assertThat(initialAliasZid).isNotBlank();
+
+                    // change non-alias property without setting alias properties to null
+                    originalIdp.setAliasId(null);
+                    originalIdp.setAliasZid(null);
+                    final IdentityProvider<?> updatedIdp = updateIdp(zone1, originalIdp);
+                    assertThat(updatedIdp.getAliasId()).isBlank();
+                    assertThat(updatedIdp.getAliasZid()).isBlank();
+
+                    // the alias IdP should have its reference removed
+                    assertReferenceWasRemovedFromAlias(initialAliasId, initialAliasZid);
+                }
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.